### ISSUE TITLE
[FEATURE] - Claim generation

### ISSUE THREAD

**Issue by cs-victor-nascimento** [bryanjos/joken#82]
While using this new API I stumbled on something we might have missed: claim generation instead of statically linking the claim. The difference is illustrated below:

```elixir
def with_claim(token, claim, value) do
  # value is always the same. Useful for `issuer` and custom claims.
end

def with_claim_generation(token, claim, function) do
  # Value is always the result of calling function on the time of signing.
  # Useful for exp, iat, nbf.
  # Can be expanded for receiving context on signing through options param.
end
```

This makes it possible for me to have a single instance of `Joken.Token` generating tokens for several requests.

What do you think?

**Comment by bryanjos** [comment]
That's a good idea! That would definitely be helpful

**Comment by cs-victor-nascimento** [comment]
Fixed in #83
